corehq/apps/cloudcare/static/cloudcare/js/formplayer/router.js

```diff
@@ -275,6 +275,12 @@ hqDefine("cloudcare/js/formplayer/router", [
     });
 
     FormplayerFrontend.on("breadcrumbSelect", function (index) {
+        if (FormplayerFrontend.formInProgress) {
+            const userConfirmed = window.confirm("You have a form in progress. Are you sure you want to navigate away?");
+            if (!userConfirmed) {
```

Comment:
I looked briefly into this the other day to see if it was possible to manually fire a `onbeforeunload` event or whatever, but I came up short.